src/Calculator/Controls/EquationTextBox.cs

```diff
@@ -0,0 +1,719 @@
+
+            }
+
+            protected override void OnPointerCaptureLost(PointerRoutedEventArgs e)
+            {
+
+                m_isPointerOver = false;
+                UpdateCommonVisualState();
+
+            }
+
+            private void OnIsAddEquationModePropertyChanged(bool oldValue, bool newValue)
+            {
+
+                UpdateCommonVisualState();
+                UpdateButtonsVisualState();
+
+            }
+
+            private void UpdateCommonVisualState()
+            {
+
+                String state = null;
+                bool richEditHasContent = RichEditHasContent();
+
+                if (m_HasFocus && HasError)
+                {
+                    state = "FocusedError";
+                }
+                else if (IsAddEquationMode && m_HasFocus && !richEditHasContent)
+                {
+                    state = "AddEquationFocused";
+                }
+                else if (m_HasFocus)
+                {
+                    state = "Focused";
+                }
+                else if (IsAddEquationMode && m_isPointerOver && !richEditHasContent)
+                {
+                    state = "AddEquation";
+                }
+                else if (HasError && (m_isPointerOver || m_isColorChooserFlyoutOpen))
+                {
+                    state = "PointerOverError";
+                }
+                else if (m_isPointerOver || m_isColorChooserFlyoutOpen)
+                {
+                    state = "PointerOver";
+                }
+                else if (HasError)
+                {
+                    state = "Error";
+                }
+                else if (IsAddEquationMode)
+                {
+                    state = "AddEquation";
+                }
+                else
+                {
+                    state = "Normal";
+                }
+                VisualStateManager.GoToState(this, state, false);
+
+            }
+
+            private void UpdateButtonsVisualState()
+            {
+
+                String state;
+
+                if (m_HasFocus && RichEditHasContent())
+                {
+                    state = "ButtonVisible";
+                }
+                else if (IsAddEquationMode)
+                {
+                    state = "ButtonHideRemove";
+                }
+                else
+                {
+                    state = "ButtonCollapsed";
+                }
+
+                VisualStateManager.GoToState(this, state, true);
+
+            }
+
+            private bool RichEditHasContent()
+            {
+                string text = null;
+                if (m_richEditBox != null)
+                {
+                    m_richEditBox.TextDocument.GetText(Windows.UI.Text.TextGetOptions.NoHidden, out text);
+                }
+                return !string.IsNullOrEmpty(text);
+
+            }
+
+            private void OnRichEditBoxGotFocus(Object sender, RoutedEventArgs e)
```

Comment:
Same as String. Could you please change the Object to object?

Reply:
Done